### PR TITLE
Add Checkstyle for formatting

### PR DESCRIPTION
**Pull request by johnoneill46**
- Introduced latest version of Checkstyle and resolved any issues it flagged
- Replaced deprecated "APPLICATION_JSON_UTF8_VALUE" with "APPLICATION_JSON_VALUE"

### CODE REVIEW

File: dependency-check-suppressions.xml [hunk 1]
```diff
@@ -18,12 +10,8 @@
...
     </suppress>
     <suppress>
-        <notes><![CDATA[
-            Ignoring app insights netty errors as it's not used ]]>
-        </notes>
-        <gav regex="true">.*</gav>
-        <cve>CVE-2014-3488</cve>
-        <cve>CVE-2015-2156</cve>
-        <cve>CVE-2019-16869</cve>
+        <notes> Suppressing unfixable vulnerabilities in latest version of Checkstyle. </notes>
```
Comment (HarryH96) [inline dependency-check-suppressions.xml]: I thought there was a fix we could do by removing the offending lines in the checkstyle file

File: build.gradle [hunk 2]
```diff
@@ -147,6 +149,25 @@ pitest {
...
 
+checkstyle {
+    maxWarnings = 0
+    toolVersion = '8.21'
+    // need to set configDir to rootDir otherwise submodule will use submodule/config/checkstyle
+    configDir = new File(rootDir, './')
+}
+
+// make build fail on Checkstyle issues (https://github.com/gradle/gradle/issues/881)
+tasks.withType(Checkstyle).each { checkstyleTask ->
```
Comment (timja) [inline build.gradle]: this hasn't been needed for ages, maxWarnings = 0 implements it, remove this whole block

File: build.gradle [hunk 3]
```diff
@@ -147,6 +149,25 @@ pitest {
     mutationThreshold = 50
 }
 
+checkstyle {
+    maxWarnings = 0
+    toolVersion = '8.21'
```
Comment (timja) [inline build.gradle]: Probably worth going to the latest? (you'll probably have to make minor changes to your checkstyle.xml
```suggestion
    toolVersion = '8.29'
```

File: build.gradle [hunk 4]
```diff
@@ -179,6 +200,7 @@ dependencies {
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: versions.jacksonCore
     compile group: 'com.google.code.findbugs', name: 'jsr305', version: versions.jsr305
     compile group: 'com.jayway.jsonpath', name: 'json-path-assert', version: versions.jaywayJsonPath
+    compile group: 'com.puppycrawl.tools', name: 'checkstyle', version: versions.puppyCrawl
```
Comment (timja) [inline build.gradle]: this shouldn't be needed

File: build.gradle [hunk 5]
```diff
@@ -147,6 +149,10 @@ pitest {
     mutationThreshold = 50
 }
 
+checkstyle.maxWarnings = 0
```
Comment (timja) [inline build.gradle]: why not use the block layout?
bit easier to read?
https://github.com/hmcts/spring-boot-template/blob/master/build.gradle#L73-L78

**Pull request merged**